components/script/dom/htmlformelement.rs

```diff
@@ -465,6 +466,15 @@ impl HTMLFormElement {
     /// https://html.spec.whatwg.org/multipage/#statically-validate-the-constraints
     fn static_validation(&self) -> Result<(), Vec<FormSubmittableElement>> {
         let node = self.upcast::<Node>();
+        for child in node.traverse_preorder() {
+            match child.downcast::<Element>() {
```

Comment:
Thank you very much for this, got a better idea about the flow now. Will soon post the changes.